pallets/nodle-staking/src/lib.rs

```diff
@@ -193,6 +195,52 @@ pub mod pallet {
             Self::deposit_event(Event::TotalSelectedSet(old, new));
             Ok(().into())
         }
+        #[pallet::weight(T::WeightInfo::set_staking_limits())]
+        pub fn set_staking_limits(
+            origin: OriginFor<T>,
+            max_stake_validators: u32,
+            min_stake_session_selection: BalanceOf<T>,
+            min_validator_bond: BalanceOf<T>,
+            min_nominator_total_bond: BalanceOf<T>,
+            min_nominator_chill_threshold: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            T::CancelOrigin::try_origin(origin)
```

Comment:
Why `CancelOrigin`?

Reply:
Good question Eliott, 
For staking config api's like 
`set_invulnerables() | set_total_validator_per_round() | slash_cancel_deferred()` 
we made design choice of origin could be either `Root or CancelOrigin`.
Here `set_staking_limits()` also falls under same class, so caller origin 
could be either `Root or CancelOrigin`.

Reply:
```suggestion
            ensure_root(origin)?;
```

Reply:
let's just `ensure_root` then